### PR TITLE
Add collections

### LINKED ISSUE #50: userStory - addCollections
## Value proposition
As a user, 
I would like to organize my recipes into collections 
in order to have a better overview.
## Description
Bildschirmfoto 2024-03-13 um 15 54 24

## Acceptance criteria
- from my user dashboard, I can access the "My Recipes" section and browse through my collections or cookbooks
- additionally, I have the option to create a new collection and select recipes to include in it
## Tasks
- [ ] create a link on the "My Recipes" page leading to a new page named `collections.js`
- [ ] introduce a collections array within the db/user. This array should store the name of the collection and all the recipe IDs it contains
- [ ] implement an "Add" button on the `collections.js` page to create a new key in the collections array
- [ ] include a "Save" button on the `recipe/index.js`. This button should store the ID of the selected recipe in a collection
- [ ] render all recipes that contains on the `collection.js`

### PR DESCRIPTION
**Pull request by avoesen**

### CODE REVIEW

File: pages/profile/collections/index.js [hunk 1]
```diff
@@ -0,0 +1,92 @@
...
+    setAddCollection(!addCollection);
+  }
+
+  function addNewCollection(event) {
+    event.preventDefault();
+    const collectionName = event.target.collectionName.value;
+    const newCollection = { collectionName: collectionName, recipes: [] };
+
+    if (!user.collections) {
+      user.collections = [{ collectionName: "", recipes: [] }];
```
Comment (gcode-de) [inline pages/profile/collections/index.js]: maybe an empty array is better here to instantiate the collections array?

File: pages/profile/collections/[id].js [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+  const router = useRouter();
+  const { id } = router.query;
+  if (!user || !recipes) {
+    return;
+  }
+
+  const foundCollection = user.collections.find((col) => col._id === id);
+
+  const foundRecipes = recipes.filter((recipe) =>
+    foundCollection.recipes.includes(recipe._id)
```
Comment (gcode-de) [inline pages/profile/collections/[id].js]: Can you use the populate-function in the user API-route instead and access the recipes in the collection directly? Maybe that's an easier and and better scalable approach?

File: pages/profile/myRecipes.js [hunk 3]
```diff
@@ -64,7 +67,17 @@ export default function MyRecipes({
...
       <Spacer />
-      <Header text={"Meine Rezepte 🥗"} />
+
+      <StyledH2>
+        <div>Kochbücher</div>
+        <StyledLink href="/profile/collections">zeig mir alle</StyledLink>
+      </StyledH2>
+      <Wrapper>
+        {user.collections.map((col, index) => (
+          <CollectionCard key={index} collection={col} />
```
Comment (gcode-de) [inline pages/profile/myRecipes.js:77]: If you don't set a limit here - do you still need the "zeig mir alle"-link?

File: public/icons/svg/notebook-alt_9795395.svg
Comment (gcode-de) [inline public/icons/svg/notebook-alt_9795395.svg:1]: I think this file needs a bigger stroke compared to the other icons.

File: pages/recipe/[id]/index.js [hunk 4]
```diff
@@ -65,12 +76,29 @@ export default function DetailPage({
...
-  }
+  async function handleCollection(event) {
+    event.preventDefault();
+    const isDuplicate = user.collections.some((col) =>
+      col.recipes.map((recipe) => recipe.id === id)
+    );
+    if (isDuplicate) {
+      notifyError("Dieses Rezept ist bereits gespeichert.");
+      return;
+    }
```
Comment (gcode-de) [inline pages/recipe/[id]/index.js]: this logic means, a recipe can only be saved to one collection per user. Is this the desired behaviour?
Haha, I doubt it!

**Review by gcode-de — CHANGES_REQUESTED** [review]
I suggested some minor changes, let's talk about them later if you want.
Overall really nice structured and modular code! :-)

File: public/icons/svg/notebook-alt_9795395.svg
Comment (avoesen) [reply public/icons/svg/notebook-alt_9795395.svg:1]: it's from the same collection, I think that's due to the scale

File: pages/profile/myRecipes.js (hunk 3 above)
Comment (avoesen) [reply pages/profile/myRecipes.js:77]: you can currently only add new collections on the collection page and that's the only way :D

**Pull request merged**